code/obj/item/go_board.dm

```diff
@@ -0,0 +1,196 @@
+						user << output("black","go.browser:checkhand")
+						user.u_equip(equipped)
+						qdel(equipped)
+						return
+					else if(istype(equipped,/obj/item/gostonew))
+						user << output("white","go.browser:checkhand")
+						user.u_equip(equipped)
+						qdel(equipped)
+						return
+				else if(usr.client && !(usr in range(usr.client.view,src)))
+					usr << browse(null, "window=go")
+					return
+			if("piecelog")
+				var/matchfound = 0
+				if(piecelist.len)
+					for(var/i=1,i<=src.piecelist.len,i++)
+						if("[src.piecelist[i]["position"]]" == "[href_list["position"]]")
+							matchfound = 1
+							break
+				if(matchfound == 0)
+					src.piecelist.Add(list(list("position"=href_list["position"],"color"=href_list["color"])))
+				var offsetx = text2num(href_list["offsetx"])
+				var offsety = text2num(href_list["offsety"])
+				var color = href_list["color"]
+
+				if(!(src.GetOverlayImage("[href_list["position"]]")))
+					var/image/piecedisplay = new /image('icons/obj/items/items.dmi',"go-render-stone-[color]")
+					piecedisplay.pixel_x = (-offsetx)
+					piecedisplay.pixel_y = (-offsety)
+
+					src.UpdateOverlays(piecedisplay,"[href_list["position"]]")
+
+				for(var/mob/living/carbon/human/u in src.openwindows)
+					if(u == usr)
+						continue
+					if(u.client && !(u in range(u.client.view,src)))
+						u << browse(null, "window=go")
+						break
+					u << output(list2params(list("null","null","[color]","null","secondary","[href_list["position"]]")),"go.browser:createpiece")
+				return
+			if("remove")
+				if(istype(usr,/mob/living/carbon/human) && (usr in range(1,src)))
+					var/mob/living/carbon/human/user = usr
+					src.ClearSpecificOverlays("[href_list["position"]]")
+
+					if(href_list["color"]=="black")
+						user.put_in_hand_or_drop(new /obj/item/gostoneb)
+					else
+						user.put_in_hand_or_drop(new /obj/item/gostonew)
+
+					for(var/i=1,i<=src.piecelist.len,i++)
+						if("[src.piecelist[i]["position"]]"==href_list["position"])
+							src.piecelist[i] = null
+							src.piecelist.Remove(null)
+					for(var/mob/living/carbon/human/u in src.openwindows)
+						if(u == user)
+							continue
+						if(u.client && !(u in range(u.client.view,src)))
+							u << browse(null, "window=go")
+							break
+						u << output("[href_list["position"]]","go.browser:removepiece")
+					return
+				else if(usr.client && !(usr in range(usr.client.view,src)))
+					usr << browse(null, "window=go")
+					return
+
+	MouseDrop(mob/user as mob)
+		if((istype(user,/mob/living/carbon/human))&&(!user.stat)&&!(src in user.contents))
+			user.put_in_hand_or_drop(src)
+
+	disposing()
+		for(var/mob/living/carbon/human/user in src.openwindows)
+			user << browse(null, "window=go")
+		..()
+
+/obj/item/gobowlb
+	name = "bowl of black stones"
+	desc = "a bowl of little black stones."
+	icon = 'icons/obj/items/items.dmi'
+	icon_state = "go-bowl-b"
+
+	attack_hand(mob/user as mob)
+		user.put_in_hand_or_drop(new /obj/item/gostoneb)
```

Comment:
I'm mildly worried about this being unlimited. Probably not too big of a problem but spacepeople are good at finding creative ways to abuse unlimited items.

Reply:
I was thinking if it becomes a problem later I can change it.

Reply:
Done in next commit